.claude/hooks/pre-edit.sh

```
@@ -0,0 +1,148 @@
+
+if [[ -z "$FILE" ]]; then
+  echo "Usage: $0 <file-path>"
+  exit 1
+fi
+
+# Normalize file path to be repo-relative for consistent pattern matching
+# This ensures patterns like vendor/* work regardless of whether the input is absolute or relative
+REPO_ROOT=$(git rev-parse --show-toplevel 2>/dev/null || echo ".")
+if [[ "$FILE" = /* ]]; then
+  # Convert absolute path to repo-relative
+  FILE="${FILE#$REPO_ROOT/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)